installer/lib/phoenix_new.ex

```diff
@@ -326,6 +329,17 @@ defmodule Mix.Tasks.Phoenix.New do
     end
   end
 
+  defp set_ecto_adapter("mssql"), do: {:tds_ecto, Tds.Ecto}
+  defp set_ecto_adapter("mysql"), do: {:mariaex, Ecto.Adapters.MySQL}
+  defp set_ecto_adapter(_),       do: {:postgrex, Ecto.Adapters.Postgres}
```

Comment:
We should do `opts[:database] || "pg"` upstream and here check for "pg". If anything else is given, we should fail with `Mix.raise`.